### DISCUSSION TITLE
Goal of the project

### DISCUSSION THREAD

**Discussion by vaibhav-kaushal** [techrail/devta#1] [Announcements]
This post is to come up with a broad goal of the project. Below is a brief that I had written earlier:

---

This is a frontend only Project. Devta is the Hindi term for a "divine being" who is helpful to people by nature. The name starts with "Dev" standing for developer and the "ta" is just a play on the intention of the tool - to be helpful for developers. The aim is to build a single tool which most developers could use for their everyday tasks. For example (but not limite to): Color Picking, Formatters for markups and languages (such as YAML, JSON, XML, HTML, JS etc.), generating Hashes (MD5, SHA, Blowfish), Sanitising and detecting errors in CSS, Minification of source code, working with time (timezone conversions, parsing time strings, UNIX timestamp to Date-Time formats, adding and subtracting times etc.), base64 encoding and decoding and so on.

Such tools already exist. Some are online and some are offline. This project is supposed to be available online as a website (responsive so that it can be used on mobile as well) as well as have electron apps for popular platforms (Linux, Mac, Windows).

PS: This project is inspired from the fantastic [DevUtils](https://devutils.com/) app which is available only for macOS as a paid tool.

**Comment by HariKrishna-28** [comment]
Interesting idea and I would love to contribute. Where do we start?

**Comment by arjun3649** [comment]
i would love to contribute too

**Comment by vaibhav-kaushal** [comment]
Thanks @arjun3649 and @HariKrishna-28 We can start with creating a sample design of how it should look and the data formats and conversions we want to support in say an Alpha release. That would help us get started. 

We already have a reference design in the DevUtils app. For now, I think we can start with UNIX time conversion and hashing of strings. Remember, this has to be delivered as both an Electron app as well as a website.

**Reply by HariKrishna-28** [reply]
Okay. Have you got any front end framework in mind for the web? And how many hashing algorithms are we implementing in the first build?

**Reply by vaibhav-kaushal** [reply]
Let's start with VueJS as the base frontend framework (and any associated frontend stuff like state manager and routers for Vue's ecosystem). For hashing, let's just stick to MD5 and SHA256 for now.

**Comment by arjun3649** [comment]
how we gonna work this out like i dont have any knowledge of Vuejs i am gonna learn it obviously  and what should be done by whom?

**Reply by vaibhav-kaushal** [reply]
Don't worry too much. All of us who want to contribute on this one need to know at least these common technologies: 

1. **HTML** - This is basic and we all need to be aware of the basics. 
2. **CSS** - It is always extremely important to know the basics. However we won't be writing things from scratch. We can make use of the _Twitter Bootstrap CSS framework_. So learning that would be great!
3. **JS** - The base JS knowledge is required to debug anything we are trying to do and to write the core functions (such as hashing and time conversions). UI interaction has to be handled using _VueJS_.

**Reply by arjun3649** [reply]
ok i understand.
